watcher/tool/watcher/main.go

```diff
@@ -82,15 +88,22 @@ func root(ccmd *cobra.Command, args []string) error {
 	ctx := context.TODO()
 	retryC := runRetryLoop(ctx)
 
+	config := watcherConfig{
+		ctx:              ctx,
+		retryC:           retryC,
+		kubernetesClient: client,
+	}
+
 	switch mode {
 	case constants.ModeDashboards:
-		err = runDashboardsWatcher(ctx, client, retryC)
+		err = runDashboardsWatcher(config)
 	case constants.ModeRollups:
-		err = runRollupsWatcher(ctx, client, influxDBConfig, retryC)
+		err = runRollupsWatcher(config, influxDBConfig)
 	case constants.ModeAlerts:
-		err = runAlertsWatcher(ctx, client, retryC)
+		err = runAlertsWatcher(config)
 	default:
-		return trace.Errorf("ERROR: unknown mode %q\n", mode)
+		return trace.BadParameter("bad mode type. Should be %v, %v or %v", constants.ModeAlerts,
```

Comment:
nit: it would be simpler if all supported modes were stored in a variable. You could then just dump it w/o explicitly listing each mode value:
```go
return trace.BadParameter("invalid mode, supported modes are %v, constants.SupportedModes)
```
another benefit is you only need to update one place if you add a mode.